ldap_auth_provider.py

```diff
@@ -88,8 +90,9 @@ def check_password(self, user_id, password):
         localpart = user_id.split(":", 1)[0][1:]
 
         try:
+            tls = ldap3.Tls(validate=ssl.CERT_REQUIRED)
```

Comment:
Just to check: this still allows you to use a plain connection right? Just that if you do use SSL it will correctly validate the cert?

Reply:
That's right, if you connect to LDAP without SSL or TLS, the Tls object is ignored.

Reply:
:tada:

Reply:
Just to document this... I tested this with a publicly accessible LDAP server:

```
import ssl
import ldap3

tls = ldap3.Tls(validate=ssl.CERT_REQUIRED)
uri = 'ldap://ldap.andrew.cmu.edu:389'
server = ldap3.Server(uri, get_info=ldap3.ALL, tls=tls)
connection = ldap3.Connection(server, auto_bind=ldap3.AUTO_BIND_NO_TLS)

connection.server.info # Output of server details (i.e. we are connected)
connection.tls_started # returns False
```